portal-impl/src/com/liferay/portlet/PortalPreferencesImpl.java

```diff
@@ -310,7 +349,12 @@ protected boolean isCausedByStaleObjectException(Throwable t) {
 		return false;
 	}
 
-	protected void retryableStore(Callable<?> callable) throws Throwable {
+	protected void retryableStore(
+			Callable<?> callable, String namespace, String key)
```

Comment:
I'll need to call _encodeKey() before super.getValues() to get the right value then. I could just bump it up to the calling method, though?